### PR TITLE
Link new IA accounts to existing, previously linked OL account

### LINKED ISSUE #8696: Attempting to link an existing account silently fails
### Evidence / Screenshot (if possible)

Screenshot 2024-01-04 at 19 37 33

Screenshot 2024-01-04 at 19 37 57

### Relevant url?
https://openlibrary.org/account/login

### Steps to Reproduce
1. Sign up to Internet Archive with the "Sign in with Google" option
2. Log into Open Library
3. Delete Internet Archive account
4. Create a new Internet Archive account with the same email address
5. Log into Open Library with the "Sign in with Google" option
6. Click the "Link accounts" button

* Actual: The linking silently fails
* Expected: To be able to relink my existing Open Library account with the new Internet Archive account

### PR DESCRIPTION
**Pull request by jimchamp**
Closes #8696

Re-establishes linkage between existing Open Library account and an IA account that has been deleted but later recreated using the same email address.

Publishes an "IA <-> OL itemname mismatch" event to sentry, which includes the old and new `itemname`s in the "Additional Details" section (as `ol_itemname` and `ia_itemname`, respectively).

### Technical

`logger.error` calls automatically publish events to Sentry (details [here](https://docs.sentry.io/platforms/python/integrations/logging/#verify)).  Such event, however, do not contain the entire stacktrace.  Additional data that is included in the `extra` dict will appear in the event's "Additional Data" section.

> [!important]
> I've only included the old and new `itemname` in these Sentry events.  Will this be enough to determine if we're only entering the `if ol_account and ol_account.itemname` code block when 2 existing accounts become unlinked?

An "IA <-> OL itemname mismatch" Sentry event can be found [here](https://sentry.archive.org/organizations/ia-ux/issues/480372/events/969c127feee347a19d23d658b1aa711c/?project=7).
The extra data can be found in this [section](https://sentry.archive.org/organizations/ia-ux/issues/480372/events/969c127feee347a19d23d658b1aa711c/?project=7#extra).

### Testing

I've tested this by doing the following:

1. Creating a new IA account with Google Sign-in.
2. Logging into OL using Google Sign-in.
3. Deleting the account on IA.
4. Creating a new IA account with Google Sign-in, using the same Google account that was used in step 1.
5. Logging into OL again using Google Sign-in.

On success, I was redirected to the My Books page.
When this fails, the login page is reloaded without an error message.

### Stakeholders

@mekarpeles 
@seabelis

### CODE REVIEW

**Comment by jimchamp** [comment]
Squash on merge.

**Comment by jimchamp** [comment]
No error message is displayed when/if Google sign-in fails.  In order to debug, removing this branch from testing for now.

File: openlibrary/accounts/model.py [hunk 1]
```diff
@@ -825,11 +825,17 @@ def audit_accounts(
         ol_account = OpenLibraryAccount.get(link=ia_account.itemname, test=test)
         link = ol_account.itemname if ol_account else None
 
-        # The fact that there is no link implies no Open Library account exists
-        # containing a link to this Internet Archive account...
+        # The fact that there is no link implies either:
+        # 1. There was no Open Library account ever linked this IA account
```
Comment (mekarpeles) [inline openlibrary/accounts/model.py:829]:
```suggestion
        # 1. There was no Open Library account ever linked to this IA account
```

**Review by mekarpeles — APPROVED** [review]
lgtm

**Pull request merged**